app/src/actioncommands.cpp

```diff
@@ -703,10 +702,9 @@ void ActionCommands::duplicateLayer()
     int currFrame = mEditor->currentFrame();
 
     Layer* toLayer = layerMgr->createLayer(fromLayer->type(), tr("%1 (copy)", "Default duplicate layer name").arg(fromLayer->name()));
-    toLayer->removeKeyFrame(1);
     fromLayer->foreachKeyFrame([&] (KeyFrame* key) {
         key = key->clone();
-        toLayer->addKeyFrame(key->pos(), key);
+        toLayer->loadKey(key);
```

Comment:
I feel like exposing the loadKey function is wrong, to me it should be an private function. It's only ever been used inside layer, it's a generic function that simply loads a keyframe into the map, however used wrongly, the keyframe could end up not being initialized correctly.

Maybe it would be better to introduce a new function for the purpose of either adding or replacing a keyframe? 

Thoughts?